File: shared/src/main/java/ru/privatenull/pnauth/config/CaptchaSettings.java
package ru.privatenull.pnauth.config;

import java.time.Duration;

public record CaptchaSettings(boolean enabled, Duration lifetime, int maxAttempts) {
    public CaptchaSettings {
        if (lifetime == null || lifetime.isZero() || lifetime.isNegative() || maxAttempts < 1) {
            throw new IllegalArgumentException("Invalid captcha settings");
        }
    }

    public static CaptchaSettings defaults() {
        return new CaptchaSettings(false, Duration.ofSeconds(30), 3);
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/config/CaptchaSettings.kt
package ru.privatenull.pnauth.config

import java.time.Duration

@JvmRecord
data class CaptchaSettings(val enabled: Boolean, val lifetime: Duration, val maxAttempts: Int) {
    init {
        if (lifetime.isZero || lifetime.isNegative || maxAttempts < 1) {
            throw IllegalArgumentException("Invalid captcha settings")
        }
    }

    companion object {
        @JvmStatic
        fun defaults(): CaptchaSettings {
            return CaptchaSettings(false, Duration.ofSeconds(30), 3)
        }
    }
}
